app/uk/gov/hmrc/fileupload/infrastructure/ReactiveMongoConnector.scala

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2016 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.fileupload.infrastructure
+
+import java.util.concurrent.TimeUnit
+
+import play.api._
+import play.api.inject.ApplicationLifecycle
+import play.modules.reactivemongo.ReactiveMongoComponent
+import reactivemongo.api.FailoverStrategy
+import uk.gov.hmrc.mongo.MongoConnector
+
+import scala.concurrent.Future
+import scala.concurrent.duration.FiniteDuration
+
+class ReactiveMongoConnector(config: play.api.Configuration, lifecycle: ApplicationLifecycle) extends ReactiveMongoComponent {
+
+  def mongoConnector: MongoConnector = _mongoConnector.getOrElse(throw new Exception("ReactiveMongoPlugin error: no MongoConnector available?"))
+
+  Logger info "ReactiveMongoPlugin starting..."
+
+  val _mongoConnector: Option[MongoConnector] = {
+
+    val mongoConfig = config.getConfig("mongodb")
```

Comment:
we will want to uncomment this line and pass in the mode before merging, this code is copied from play-reactivemongo hmrc library, in this state our configuration will potentially not work in the various environments.